fiftyone/core/dataset.py

```diff
@@ -2293,27 +2295,91 @@ def save_view(self, name, view, overwrite=False):
         if view._root_dataset is not self:
             raise ValueError("Cannot save view into a different dataset")
 
-        if name in self._doc.views:
+        if self.has_view(name):
             if not overwrite:
                 raise ValueError(
                     "Saved view with name '%s' already exists" % name
                 )
 
-            _view_doc = self._doc.views.pop(name)
+            _view_doc = self._get_view_doc(name, pop=True)
             _view_doc.delete()
 
+        now = datetime.utcnow()
         view_doc = foo.ViewDocument(
             dataset_id=self._doc.id,
+            name=name,
+            url_name=fou.to_url_name(name),
+            description=description,
+            color=color,
             view_stages=[
                 json_util.dumps(s)
                 for s in view._serialize(include_uuids=False)
             ],
+            index=len(self._doc.views),
+            created_at=now,
+            last_modified_at=now,
         )
         view_doc.save()
 
-        self._doc.views[name] = view_doc
+        self._doc.views.append(view_doc)
         self._doc.save()
 
+    def get_view_info(self, name):
+        """Loads the editable information about the saved view with the given
+        name.
+
+        Args:
+            name: the name of a saved view
+
+        Returns:
+            a dict of editable info
+        """
+        view_doc = self._get_view_doc(name)
+        return {f: view_doc[f] for f in view_doc._EDITABLE_FIELDS}
+
+    def update_view_info(self, name, info):
+        """Updates the editable information for the saved view with the given
+        name.
+
+        Examples::
+
+            import fiftyone as fo
+            import fiftyone.zoo as foz
+
+            dataset = foz.load_zoo_dataset("quickstart")
+            view = dataset.limit(10)
+
+            dataset.save_view("test", view)
+
+            info = dataset.get_view_info("test")
+            info["name"] = "a new name"
+
+            dataset.update_view_info("test", info)
+
+        Args:
+            name: the name of a saved view
+            info: a dict whose keys are a subset of the keys returned by
+                :meth:`get_view_info`
+        """
+        view_doc = self._get_view_doc(name)
```

Comment:
is it worth checking if the view exists?

Reply:
That logic is now in `_get_view_doc()`